src/provisioner/cluster.py

```diff
@@ -149,7 +148,8 @@ def save(self, cortx_conf):
                 if socket.getfqdn() == node['hostname']:
                     with open(const.MACHINE_ID_PATH,'w') as machine_id_path:
                         machine_id_path.write(machine_id)
-                    if node['type'] in Const.NODE_TYPE_DATA.value:
+                        storage_spec = node.get('storage')
+                    if storage_spec:
```

Comment:
since we are not using storage_spec anywhere else in this scope we may directly use 
if node.get('storage'):